### PR TITLE
History entity ids iteration

### PR DESCRIPTION
**Pull request by Kadrian**

### CODE REVIEW

File: frontend/src/js/entity-history/LoadHistoryDropzone.tsx [hunk 1]
```diff
@@ -46,22 +48,26 @@ export const LoadHistoryDropzone = ({
...
+
+      if (row.length > 2) {
+        loadedEntityStatus[row[1]] = row
+          .slice(2)
+          .filter((str) => str.length > 0)
+          .map((s) => {
+            const opt = s.trim();
+            loadedEntityStatusOptionsRaw.push(opt);
+            return { label: opt, value: opt };
+          });
```
Comment (awildturtok) [inline frontend/src/js/entity-history/LoadHistoryDropzone.tsx:70]: This section would benefit from an explanation of what `data` is and why you're preferably looking for rows of exactly length 2.

File: frontend/src/js/entity-history/Navigation.tsx [hunk 2]
```diff
@@ -104,17 +104,22 @@ export const Navigation = memo(
...
-
-      const csvString = entityIds
-        .map(idToRow)
-        .map((row) => row.join(";"))
-        .join("\n");
+      const usedStatuses = Object.values(entityIdsStatus).reduce(
+        (longest, el) => (longest.length > el.length ? longest : el),
+        [],
+      );
+      const idToRow = (entityId: EntityId) =>
```
Comment (awildturtok) [inline frontend/src/js/entity-history/Navigation.tsx:111]: Why not simply use Papa.unparse for this?

**Review by awildturtok — APPROVED** [review]
Neither of these suggestions is a dealbreaker

File: frontend/src/js/entity-history/LoadHistoryDropzone.tsx (hunk 1 above)
Comment (Kadrian) [reply frontend/src/js/entity-history/LoadHistoryDropzone.tsx]: Good point, I grouped the save and load logic + added a comment

File: frontend/src/js/entity-history/Navigation.tsx (hunk 2 above)
Comment (Kadrian) [reply frontend/src/js/entity-history/Navigation.tsx]: yeah good point, done!

**Pull request merged**